lib/src/kbasesearchengine/main/IndexerWorker.java

```diff
@@ -477,7 +478,8 @@ private void indexObject(
         long t1 = System.currentTimeMillis();
         final File tempFile;
         try {
-            tempFile = ObjectParser.prepareTempFile(getTempSubDir(guid.getStorageCode()));
+            FileUtil.getOrCreateCleanSubDir(rootTempDir, guid.getStorageCode());
```

Comment:
Deleting the subdir seems like a real problem here - firstly, it means that the workers can't be threaded, which is one of the items on the search priority list. Secondly, if I understand how things work with the indexing recursion, this will delete the files produced higher up in the recursion.